app/script/ui/viewportObserver.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+const observedElements = new Map();
+const onIntersect = entries => {
+  entries.forEach(({isIntersecting, target: element}) => {
+    if (isIntersecting) {
+      const onElementIntersects = observedElements.get(element);
+      _removeElement(element);
+      if (onElementIntersects) {
+        onElementIntersects();
+      }
+    }
+  });
+};
+
+const options = {root: null, rootMargin: '0px', threshold: 0.0};
+const fullyInViewObserver = new IntersectionObserver(onIntersect, Object.assign({}, options, {threshold: 1.0}));
```

Comment:
This is the acual behavior fix. To only see the diff of this fix, please have a look at this specific commit https://github.com/wireapp/wire-webapp/pull/5319/commits/412f2a1249c8d94f5222bd141de386a153ab0f23

The rest is only creating module out of existing utils